pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -223,11 +226,10 @@ def download_binaries(self, force):
             os.chmod(b['system_path'], 0o755)
             logging.info(f'downloaded binary from {b["url"]}')
 
-            with self.lock:
-                self.data['binaries'].append(b)
-                if self.data['current_neard_path'] is None:
-                    self.reset_current_neard_path()
-                self.save_data()
+            self.data['binaries'].append(b)
+            if self.data['current_neard_path'] is None:
+                self.reset_current_neard_path()
+            self.save_data()
```

Comment:
Why are we getting rid of `self.lock` here? I though that every time you save data, you need to take a lock.

Reply:
i split this pr up into a couple commits so hopefully it would be easier to see whats going on. if you look at this first commit, the commit message kind of explains what i wanted to achieve here. basically just moving this locking to the caller of this function, because i think it is kind of unsound to lock and unlock it inside this function